File: 1803-average-waiting-time/average-waiting-time.py
class Solution:
    def averageWaitingTime(self, customers: List[List[int]]) -> float:
        total_wait_time = 0
        current_time = 0
        
        for arrival_time, prep_time in customers:
            if current_time < arrival_time:
                current_time = arrival_time
            
            wait_time = current_time - arrival_time + prep_time
            total_wait_time += wait_time
            
            current_time += prep_time
        
        return total_wait_time / len(customers)

File: 1803-average-waiting-time/average-waiting-time.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        long totalWaitTime = 0;
        int currentTime = 0;
        
        for (int[] customer : customers) {
            int arrivalTime = customer[0];
            int prepTime = customer[1];
            
           
            if (currentTime < arrivalTime) {
                currentTime = arrivalTime;
            }
            
          
            int waitTime = currentTime - arrivalTime + prepTime;
            totalWaitTime += waitTime;
            
    
            currentTime += prepTime;
        }
        

        return (double) totalWaitTime / customers.length;
    }
}